### ISSUE TITLE
Better live docs

### ISSUE THREAD

**Issue by fonsp** [fonsp/Pluto.jl#111]
Some improvements:

- [x] Show what is currently being searched for
- [x] Support cross-reference links between docs
- [ ] Show that it is searching
- [ ] Title of the docs should be a textbox where you can enter a query directly
- [x] Handle dots, like `Module.funky`
- [ ] Function arguments
- [ ] Show when the name is found but it has no docs
- [ ] To the right of a cell might make more sense
- [ ] Heuristics for which function is dispatched on - can VS Code do this nowadays? I guess that we can do this dynamically, since some of the types are known

Extras:

- [ ] `apropos(str)` might be useful. Is resource intensive though, maybe write our own version that just returns the first hit? Although that might not be the most relevant one
- [ ] Show filename and line number of defintion. Double cool if it is a clickable link to GitHub! (Like in methodtables shown as `text/html`) I found this useful when the docs are unclear or when you want to know how something works. 

What is Julia's doc system missing?
- [ ] Given an object, **tell you what you can do with it**. In OOP languages this is `obj.<TAB>` to autocomplete.

### Please post other ideas and suggestions!

**Comment by ilyagr** [comment]
> Given an object, tell you what you can do with it. In OOP languages this is obj.<TAB> to autocomplete.

The simplest thing to do is to show the docs for `typeof(x)` when there are no useful docs for `x`.

The next step might be to find a way to show `methodswith(typeof(x); [supertypes=true])` together with the docs for `typeof(x)`. The output is kind of long to put into the Live Docs screen, but if you make it a tree based on which module different methods come from, it could be OK. 

One problem is that adding `supertypes=true` is helpful for some types and harmful for others. Perhaps you could show `supertypes(typeof(x))` if it's non-trivial. If the user clicks on one of the supertypes, you could show `methodswith` for that supertype.

**Comment by fonsp** [comment]
Thanks! Perhaps as an experiment, you could take an existing notebook, and run the methods that @ilyagr suggested and see if the results would be helpful.

**Comment by fonsp** [comment]
About autocompleting functions that can be dispatched on something: https://discourse.julialang.org/t/why-is-multiple-dispatch-a-feature/31101/10?u=pfitzseb

**Comment by Avinash-Bhat** [comment]
@fonsp an improvement that could benefit when using a mobile/tablet device is that the live-docs is hidden for a tablet screen sizze, but it may be shown as a bottom sheet which can help in getting a quick look at the docs even in tablet/mobile device.

#### Off Topic: How this affects my work: 

On Firefox, I use a 75%:25% tab using the [SideView](https://addons.mozilla.org/en-US/firefox/addon/side-view/) extension, I can use the 75% part for the notebook, and the 25% for reference material (Videos, Lecture Notes...), using live-docs is impossible in this scenario.

**Comment by fonsp** [comment]
I felt like screen space is much more valuable on small devices, which is why it is disabled. But it sounds like a normal sized screen is consider "too small" by Pluto?

**Comment by Avinash-Bhat** [comment]
@fonsp I think the reason it consider as "too small" is from the calculated value at [frontend/editor.css#L1104](https://github.com/fonsp/Pluto.jl/blob/b53d282a132752e34563593b8c7c294866042efb/frontend/editor.css#L1104)

>  `@media screen and (min-width: calc(700px + 25px + 6px + 350px)) {`

I believe using a px based calculation makes it device dependent. (I believe you are using a Mac). Although it can be refactored to use device independent unit or percentage, I'm not sure if pursuing it would be fixing this issue.

Even if screen is "too small", I think the documentation could be quite useful despite the device type. it will be just a matter of identifying how the documentation can be shown in accessible way despite the device type.

**Comment by fonsp** [comment]
I'm not sure what you mean with device (in)dependent. `1px` does not correspond to a physical pixel in CSS, but to roughly 1/96th of 1 physical inch, right?

**Comment by Avinash-Bhat** [comment]
yes, you are correct, but my point is not on how the screen size is chosen. but rather on the user experience of the live docs

**Comment by fonsp** [comment]
(Hiding the above conversation because it is resolved by https://github.com/fonsp/Pluto.jl/commit/744e0029d1270e7893459c9256b74855b38eafdb)

**Comment by lungben** [comment]
One more wish:
Currently, when the notebook is busy executing a cell, the Live Docs are not displayed. It would be great if the Live Docs are always available, so that I can use the time waiting for a cell to be finished to look up info for writing the next cell.

**Comment by fonsp** [comment]
I don't think that we can get docs if the process is busy.

**Comment by KristofferC** [comment]
https://github.com/JuliaLang/julia/pull/38791 is related.

**Comment by fonsp** [comment]
# New discussion page

We created https://github.com/JuliaCommunity/live-docs-discussion/discussions as a dedicated place to discuss Pluto's Live Docs feature, and future ideas for Julia's documentation in general! Take a look!

**Comment by danvinci** [comment]
Idea: what about a simple go-back/go-forward navigation for the live docs?

(I didn't see listed here, in the discussion page, or among the issues).

I often go down 2-3 levels in the documentation and when I'm done understanding something I want to go back to where I started. Might be already very cool having just 3-4 steps saved.

I'm willing to look into it with some pointers and occasional hand-holding.
